### ISSUE TITLE
Fine-tuning SRGAN 

### ISSUE THREAD

**Issue by hosseinahm1367** [mseitzer/srgan#9]
I am trying to fine-tune the SRGAN network initializing it with the pre-trained SRGAN checkpoint provided in the source code. I tried to do same way you suggested previously: https://github.com/mseitzer/srgan/issues/8

Basically, i simply replaced srresnet.pth with srgan.pth in the srgan config file. I get a model loading error in "load_model_state_dict" function. How should I approach this?
Thanks,

**Comment by hosseinahm1367** [comment]
Another issue I ran into recently: 
I tried training the Resent with scale factor of 8x (by changing the upscale_factor in srresnet.json and resnet.jason). However, I get this error related to target and input not having the same size: "RuntimeError: input and target have different number of elements: input[16 x 3 x 192 x 192] has 1769472 elements, while target[16 x 3 x 96 x 96] has 442368 elements"
I went into the code, and printed the size of the SRResnet network prediction as,
    print(inp.size())
    print(target.size())
    print(prediction.size()) 
and I got these:
torch.Size([16, 3, 12, 12])
torch.Size([16, 3, 96, 96])
torch.Size([16, 3, 192, 192]).

Do you have any suggestion on where could be the issue? in the up-sampling scheme maybe?

**Comment by mseitzer** [comment]
Yeaah, this is a bit more complicated (I have never done it this way around, but it should work). Note that `"pretrained_weights"` in the config has two arguments, the path to the checkpoint, and a key (in this case, `model`). The key describes under which key in the checkpoint dictionary the loader should look for the model weights. 

For normal training, this is simply `model`. The checkpoint is created here: https://github.com/mseitzer/srgan/blob/70f88fbac26b5a3659847965befea2fdd49eb625/training/runner.py#L105-L109
For adversarial training, the checkpoint has a different structure, and the model key is `generator`. It is created here: https://github.com/mseitzer/srgan/blob/70f88fbac26b5a3659847965befea2fdd49eb625/training/adversarial_runner.py#L183-L189

Basically, you want to replace `model` with `generator`, and it should work:

```
"model": {
  "#include": "resnet.json",
  "pretrained_weights": ["../resources/pretrained/srgan.pth", "generator"]
}
```
